projects/flow.groups-2D-random-walk/src/project.py

```diff
@@ -0,0 +1,211 @@
+"""Define the operations for the 2D Gaussian Random Walk project."""
+
+import flow
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# Helper functions
+def get_replica(jobs, replica):
+    """Get a certain replica from a group of jobs."""
+    try:
+        job = next(filter(lambda job: job.sp.replica == replica, jobs))
+    except StopIteration:
+        raise RuntimeError(f"No replica {replica} found.")
+    return job
+
+
+def generate_stores(jobs, store_name):
+    """Yield a data store for each job in jobs."""
+    for job in jobs:
+        with job.data:
+            yield job.data[store_name]
+
+
+class RandomWalkProject(flow.FlowProject):
+    """Create a workflow for simulating 2D Gaussian random walks."""
+
+    pass
+
+
+@RandomWalkProject.label
+def simulated(job):
+    """Return whether the job simulated."""
+    return "positions" in job.data
+
+
+def all_simulated(*jobs):
+    """Return whether all jobs simulated."""
+    return all(simulated(job) for job in jobs)
+
+
+# Operations on individual jobs
+
+# Group for operations on single jobs that are reachable from simulate
+base = RandomWalkProject.make_group("base")
+
+
+@base
+@RandomWalkProject.post(simulated)
+@RandomWalkProject.operation
+def simulate(job):
+    """Simulate a 2D random walk."""
+    # use job id as random seed
+    n_steps = job.doc.run_steps
+    generator = np.random.default_rng(int(job.id, 16))
+    # Generate all moves at once since they are independent
+    moves = generator.normal(job.sp.mean, job.sp.standard_deviation, (n_steps, 2))
+    position = np.zeros((n_steps + 1, 2), dtype=float)
+    # Store the results: cumsum aggregates all previous moves from the origin
+    position[1:] = np.cumsum(moves, axis=0)
+    job.data.positions = position
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post.true("radius_of_gyration")
+@RandomWalkProject.operation
+def compute_radius_of_gyration(job):
+    """Compute the radius of gyration for a random walk."""
+    with job.data:
+        positions = job.data["positions"][:]
+    avg_position = positions.mean(axis=0)
+    deviation = positions - avg_position
+    job.doc.radius_of_gyration = np.sqrt(
+        np.sum(deviation * deviation) / deviation.shape[0]
+    )
+    job.doc.average_position = avg_position
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post.true("end_to_end")
+@RandomWalkProject.operation
+def compute_end_to_end_distance(job):
+    """Compute the radius of gyration for a random walk."""
+    with job.data:
+        positions = job.data["positions"]
+        job.doc["end_to_end"] = np.linalg.norm(positions[-1] - positions[0])
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post(lambda job: "squared_distance" in job.data)
+@RandomWalkProject.operation
+def compute_squared_distance(job):
+    """Compute the squared distance for a random walk."""
+    with job.data:
+        positions = job.data.positions[:]
```

Comment:
Let's be consistent with whether we use attribute access (`job.data.positions`) or key access (`job.data["positions"]`).